legacy/reqcounter/reqcounter.go

```diff
@@ -24,28 +24,71 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// RequestCounter records the number of HTTP requests to GCR.
-type RequestCounter struct {
-	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
-	requests uint64     // Number of HTTP requests since recording started.
-	since    time.Time  // When the current counter began recording requests.
+// requestCounter records the number of HTTP requests to GCR.
+// TODO: @tylerferrara in the future, add a field 'persistent bool' to determine if
+// the counter should ever reset.
+type requestCounter struct {
+	mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	requests uint64        // Number of HTTP requests since recording started.
+	since    time.Time     // When the current counter began recording requests.
+	interval time.Duration // The duration of time between each log.
 }
 
 // log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+func (rc *requestCounter) log() {
 	// Hold onto the lock when reading & writing the counter.
 	rc.mutex.Lock()
 	defer rc.mutex.Unlock()
 
 	// Log the number of requests within this measurement window.
-	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("2006-01-02 15:04:05"), rc.requests)
+	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", rc.since.Format("2006-01-02 15:04:05"), rc.requests)
```

Comment:
It would help to log the `interval` field here, so that the user doesn't have to do time math to figure out what the temporal delta is between `rc.since` and the time that this message was logged.

Reply:
On it!